index.js

```diff
@@ -105,16 +113,28 @@ class PearIPC extends ReadyResource {
     this.stream.on('close', onclose)
 
     this._register()
+
+    if (this.server === null && this.id === -1) {
+      await this._beat()
+      this._heartbeat = setInterval(() => {
+        this._beat()
+      }, HEARTBEAT_INTERVAL)
```

Comment:
(detail) A more testable pattern is storing `this.heartbeatInterval` as a property, and having it default to `HEARTBEAT_INTERVAL`. That way the test can specify a much lower heartbeatInterval than we use in practice

Reply:
I've avoided doing that so far to ensure that I'm testing exactly what I think I am - imo as long as the interval is reasonable for the use case and not too high for the tests then it's better not to introduce the conditional. The other thing with this particular case is that we can't lower it too much, because clockdrift (and machine performance) have to be factored in for this timeout-based test to be reliable 

not opposed to doing it once we land and it's out there and working, but abundance of caution in this area makes me resist all chances of variability at the moment